engine/src/test/java/io/zeebe/engine/processing/streamprocessor/ReplayStateTest.java

```diff
@@ -154,12 +154,14 @@
                     .startEvent()
                     .callActivity("call-child", b -> b.zeebeProcessId(PROCESS_CHILD_ID))
                     .boundaryEvent("timer", b -> b.cancelActivity(true))
-                    .timerWithDuration("PT0S")
+                    .timerWithDuration("PT1M")
                     .endEvent("end")
                     .done())
             .withExecution(
                 engine -> {
                   final long piKey = engine.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+                  RecordingExporter.jobRecords(JobIntent.CREATED).await();
+                  engine.getClock().addTime(Duration.ofMinutes(1));
```

Comment:
I don't understand this. Wouldn't this be flaky? You risk increasing the time before the timer is scheduled to be triggered. Here you want to wait until `Timer.CREATE` was processed (note: not until `CREATED` was exported, as it may be exported before `Timer.CREATE` is completely processed, and before the time has been scheduled).

Reply:
ah, that's interesting. How do you want for this then? Anyways, what we do now is waiting for the job to be created to make sure the execution inside the child is deterministic.

Reply:
Ah, the job comes from inside the child? Does that guarantee us that the timer is scheduled and ready to be triggered? I just ask since we ran into several race conditions when triggering timers.

If you think that it's sure that the timer is scheduled at this point (which I expect if the service task is an element inside the call activity), then it's fine :+1: